tensorflow_similarity/evaluators/memory_evaluator.py

```diff
@@ -273,117 +264,156 @@ def calibrate(
-        for k in thresholds.keys():
-            cutpoints["optimal"][k] = thresholds[k][best_idx]
+        return optimal_cp
 
-        # reverse the threshold so they go from left to right as user expect
-        for k in thresholds.keys():  # this syntax is need for mypy ...
-            thresholds[k].reverse()
+    def _target_cutpoints(
+        self,
+        metrics: Dict[str, np.ndarray],
+        calibration_metric: ClassificationMetric,
+        target_name: str,
+        target_value: float,
+    ) -> Dict[str, Union[str, float]]:
+        """Compute the distance at the target metric for the calibration metric.
+
+        Args:
+            metrics: A mapping from metric name to a list of metric values
+              computed for each unique distance observed in the calibration
+              dataset. This dict should also include a distance key mapping
+              to the list of observed distances. We expect the distances to be
+              ascending, and for all lists to be the same length where each
+              index in the list corresponds to a distance in the distance list.
+            calibration_metric: The ClassificationMetric used for calibration.
+            target_name: The name for the target cutpoint.
+            target_value: The target metric value.
+
+        Returns:
+            A Dict of the metric values at the calibrated distance. This also
+              includes the cutpoint name, the distance, and the value of the
+              calibration metric.
+
+              ```
+              {
+                  'name': '0.90',     # Target cutpoint name.
+                  'value': 0.901,     # Closest metric value at or above the
+                                      # target cutpoint, assuming we are
+                                      # maximizing the calibration metric.
+                  'distance': 0.1,    # Calibrated distance.
+                  'precision': 0.901, # Here, we calibrated using precision.
+                  'f1': 0.4,          # We also computed F1 at this point.
+              }
+              ```
+        """
+        indicators = np.where(metrics[calibration_metric.name] >= target_value)[
+            0
+        ]
+        target_cp: Dict[str, Union[str, float]] = {}
+
+        if indicators.size > 0:
+            if calibration_metric.increasing:
+                # Take the first index above the target if the metric is increasing
+                idx = indicators[0]
+            else:
+                # Take the last index above the target if the metric is decreasing
+                idx = indicators[-1]
+
+            target_cp["name"] = target_name
+            target_cp["value"] = metrics[calibration_metric.name][idx].item()
+            for metric_name in metrics.keys():
+                target_cp[metric_name] = metrics[metric_name][idx].item()
+
+        return target_cp
+
+    def _last_argmax(self, x):
```

Comment:
nit: _last_argmax, _last_argmin, _is_lower don't have type hints

Reply:
Thanks! I added the typing.